pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -446,9 +447,15 @@ func (s *Reconciler) Exists(ctx context.Context) (bool, error) {
 
 	switch machinev1.AzureVMState(*vm.ProvisioningState) {
 	case machinev1.VMStateDeleting:
-		return true, fmt.Errorf("vm for machine %s has unexpected 'Deleting' provisioning state", s.scope.Machine.GetName())
+		klog.Infof("vm for machine %s has unexpected 'Deleting' provisioning state", s.scope.Machine.GetName())
+		return false, &apierrors.UnexpectedObjectError{
+			Object: s.scope.Machine,
+		}
```

Comment:
Are you sure we want to change this for deleting? I can see the merit in changing it for Failed, but does this then change the behaviour of the intended deletion cycle of the VM? Eg, does the machine actually get removed before the VM is removed in the cloud now?

Reply:
Yes, you are right. The card is also mentioning only issue with Failed. Reverting this change and keeping only the updated version for Failed.